actions/bump-and-release-cloud-plugin/action.yaml

```diff
@@ -0,0 +1,320 @@
+        echo "archive-checksum=${GRAFANA_PLUGIN_ARTIFACT_CHECKSUM}" >> $GITHUB_OUTPUT
+        echo "archive-latest=${GRAFANA_PLUGIN_ARTIFACT_LATEST}" >> $GITHUB_OUTPUT
+        echo "archive-latest-checksum=${GRAFANA_PLUGIN_ARTIFACT_LATEST_CHECKSUM}" >> $GITHUB_OUTPUT
+
+    # confirm that the version in the plugin archive will match the tag we are on
+    - name: Check package version
+      shell: bash
+      run: if [ "v${{ steps.metadata.outputs.plugin-version }}" != "${{ steps.output-version.outputs.version-tag }}" ]; then printf "\033[0;31mPlugin version doesn't match tag name\033[0m\n"; exit 1; fi
+
+    - name: Package plugin
+      id: package-plugin
+      shell: bash
+      run: |
+        mv dist ${{ steps.metadata.outputs.plugin-id }}
+        zip ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.plugin-id }} -r
+        md5sum ${{ steps.metadata.outputs.archive }} > ${{ steps.metadata.outputs.archive-checksum }}
+
+        # move assets to upload folder for mass upload
+        mkdir ${{ runner.temp }}/plugin-artifacts
+        mv ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.archive-checksum }} ${{ runner.temp }}/plugin-artifacts/
+
+    - name: Create latest zip
+      id: latest-zip
+      if: inputs.release-level != 'prerelease'
+      shell: bash
+      run: |
+        cd ${{ runner.temp }}/plugin-artifacts
+        cp ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.archive-latest }}
+        md5sum ${{ steps.metadata.outputs.archive-latest }} > ${{ steps.metadata.outputs.archive-latest-checksum }}
+        cd ..
+
+    - name: Validate plugin with validator config
+      if: inputs.validator-config-path != ''
+      shell: bash
+      run: |
+        npx -y @grafana/plugin-validator@latest -config ${{ inputs.validator-config-path }} ${{ runner.temp }}/plugin-artifacts/${{ steps.metadata.outputs.archive }}
+
+    - name: Validate plugin
+      if: inputs.validator-config-path == ''
+      shell: bash
+      run: |
+        npx -y @grafana/plugin-validator@latest ${{ runner.temp }}/plugin-artifacts/${{ steps.metadata.outputs.archive }}
+
+    # Now that we've validated our plugin we can safely commit the version bump
+    - name: Commit version bump
+      id: commit-version-bump
+      uses: grafana/github-api-commit-action@ccf9b520c5698380ad3b9619c5add427369b7ef1 # v0.2.0
+      with:
+        token: ${{ inputs.github-token }}
+        commit-message: 'chore(version): bump to version [${{ steps.output-version.outputs.version }}] (${{ inputs.release-level }}) [BOT]'
+
+    - name: Create tag
+      uses: rickstaa/action-create-tag@a1c7777fcb2fee4f19b0f283ba888afa11678b72 # v1.7.2
+      with:
+        commit_sha: ${{ steps.commit-version-bump.outputs.commit-sha }}
+        github_token: ${{ inputs.github-token }}
+        message: "chore(tag): tag version ${{ steps.output-version.outputs.version-tag }} [BOT]"
+        tag: ${{ steps.output-version.outputs.version-tag }}
```

Comment:
if a plugin want to do bump and tag in 2 separate steps I would trigger my release creating the tag manually but this action forces me to do the release on your way all in one with not too much flexibility